internal/tui/models.go

```diff
@@ -28,21 +35,90 @@ type Prompt interface {
 	Default() interface{}
 }
 
-// Validator function type for validation
-type Validator func(interface{}) error
+// ValidatorFunc function type for validation
+type ValidatorFunc func(interface{}) error
+
+// ANSI 16 color constants
+const (
+	ansiBlack         = "0"
+	ansiRed           = "1"
+	ansiGreen         = "2"
+	ansiYellow        = "3"
+	ansiBlue          = "4"
+	ansiMagenta       = "5"
+	ansiCyan          = "6"
+	ansiWhite         = "7"
+	ansiBrightBlack   = "8" // Gray
+	ansiBrightRed     = "9"
+	ansiBrightGreen   = "10"
+	ansiBrightYellow  = "11"
+	ansiBrightBlue    = "12"
+	ansiBrightMagenta = "13"
+	ansiBrightCyan    = "14"
+	ansiBrightWhite   = "15"
+)
+
+// colorSupported checks if color output is supported based on environment variables
+func colorSupported() bool {
+	// Check NO_COLOR environment variable (https://no-color.org/)
+	if os.Getenv("NO_COLOR") != "" {
+		return false
+	}
```

Comment:
Nit. It looks like lipgloss uses [`temenvs.EnvNoColor()`](https://pkg.go.dev/github.com/muesli/termenv#EnvNoColor), that seems to take other variables into account.
```suggestion
	if termenv.EnvNoColor() {
		return false
	}
```